crates/cast/src/starknet_commands/account/add.rs

```diff
@@ -0,0 +1,82 @@
+use crate::starknet_commands::account::{prepare_account_json, write_account_to_accounts_file};
+use anyhow::{ensure, Result};
+use camino::Utf8PathBuf;
+use cast::helpers::response_structs::AccountAddResponse;
+use cast::helpers::scarb_utils::CastConfig;
+use clap::Args;
+use starknet::core::types::FieldElement;
+use starknet::signers::SigningKey;
+
+#[derive(Args, Debug)]
+#[command(about = "Add an account to the accounts file")]
+pub struct Add {
+    /// Address of the account
+    #[clap(short, long)]
+    pub address: FieldElement,
+
+    /// Class hash of the account
+    #[clap(short, long)]
+    pub class_hash: Option<FieldElement>,
+
+    /// Account deployment status
+    #[clap(short, long)]
+    pub deployed: bool,
+
+    /// Account private key
+    #[clap(long)]
+    pub private_key: FieldElement,
+
+    /// Account public key
+    #[clap(long)]
+    pub public_key: Option<FieldElement>,
+
+    /// Salt for the address
+    #[clap(short, long)]
+    pub salt: Option<FieldElement>,
+
+    /// If passed, a profile with corresponding data will be created in Scarb.toml
+    #[clap(long)]
+    pub add_profile: bool,
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn add(
+    config: &CastConfig,
+    path_to_scarb_toml: &Option<Utf8PathBuf>,
+    chain_id: FieldElement,
+    add: &Add,
+) -> Result<AccountAddResponse> {
+    let private_key = &SigningKey::from_secret_scalar(add.private_key);
+    if let Some(public_key) = &add.public_key {
+        ensure!(
+            public_key == &private_key.verifying_key().scalar(),
+            "public key mismatch"
+        );
+    }
```

Comment:
What if we just calculated public key from private key?

Reply:
The argument is to allow passing the public key, but it's being asserted that the public key is actually from the private key. The public key is optional to pass.

Reply:
yup. but what if we just simplify things, and just calculate the key? then we can get rid of the parameter, and still make sure the public key is always there

Reply:
What prompts me to leave it is that currently all the keys that are in a single account in our accounts file, are present as a flag and available to pass. So if someone were to use the account json with the "public_key" field in there, it wouldn't throw errors, and they wouldn't need to modify the dict 🤔

Reply:
Also it's some layer that just assures the user the public and private key they passed correspond to each other

Reply:
I see your point, I just wanted to make sure UX is the best possible. Maybe we could make this flag optional, and if it's not present calculate it?

>So if someone were to use the account json with the "public_key" field in there, it wouldn't throw errors, and they wouldn't need to modify the dict 🤔

not really sure what you mean by that

Reply:
> Maybe we could make this flag optional, and if it's not present calculate it?

That's exactly what is happening right now 😄


> > So if someone were to use the account json with the "public_key" field in there, it wouldn't throw errors, and they wouldn't need to modify the dict 🤔

> not really sure what you mean by that

I thought that if someone were to take the whole account from the openzeppelin_accounts.json file, and automate it to pass the contents to `account add` it wouldn't complain about public_key being an unknown key

Reply:
>That's exactly what is happening right now 😄

Perhaps I'm blind, but I cannot see where we actually calculate the key here :P I can see that we try to verify it. Where is it done?

>it wouldn't complain about public_key being an unknown key

sure, but what you mean by "it"? I don't think our nor someone's code should do that, if we calculate the key